python/cuspatial/cuspatial/tests/spatial/nearest_points/test_nearest_points.py

```diff
@@ -0,0 +1,206 @@
+        points, linestrings
+    )
+    expected = gpd.GeoDataFrame(
+        {
+            "point_geometry_id": [0],
+            "linestring_geometry_id": [0],
+            "segment_id": [0],
+            "geometry": gpd.GeoSeries([Point(1, 1)]),
+        }
+    )
+    assert_geodataframe_equal(
+        result.to_pandas(), expected, check_dtype=False, check_index_type=False
+    )
+
+
+def test_multipoints():
+    points = cuspatial.GeoSeries(
+        [MultiPoint([(0, 0), (0.5, 0.5), (1.0, 0.5)])]
+    )
+    linestrings = cuspatial.GeoSeries([LineString([(2, 2), (1, 1)])])
+
+    result = cuspatial.pairwise_point_linestring_nearest_points(
+        points, linestrings
+    )
+    expected = gpd.GeoDataFrame(
+        {
+            "point_geometry_id": [2],
+            "linestring_geometry_id": [0],
+            "segment_id": [0],
+            "geometry": gpd.GeoSeries([Point(1, 1)]),
+        }
+    )
+    assert_geodataframe_equal(
+        result.to_pandas(), expected, check_dtype=False, check_index_type=False
+    )
+
+
+def test_multilinestrings():
+    points = cuspatial.GeoSeries([Point(0.5, 0.5)])
+    linestrings = cuspatial.GeoSeries(
+        [
+            MultiLineString(
+                [
+                    [(2, 2), (1, 1)],
+                    [(1, 1), (2, 0)],
+                    [(2, 0), (3, 1)],
+                    [(3, 1), (2, 2)],
+                ]
+            )
+        ]
+    )
+
+    result = cuspatial.pairwise_point_linestring_nearest_points(
+        points, linestrings
+    )
+    expected = gpd.GeoDataFrame(
+        {
+            "point_geometry_id": [0],
+            "linestring_geometry_id": [0],
+            "segment_id": [0],
+            "geometry": gpd.GeoSeries([Point(1, 1)]),
+        }
+    )
+    assert_geodataframe_equal(
+        result.to_pandas(), expected, check_dtype=False, check_index_type=False
+    )
+
+
+def test_random_input_nearest_point(
+    multipoint_generator, multilinestring_generator
+):
+    num_pairs = 10
+    max_num_points_per_multipoint = 10
+    max_num_linestring_per_multilinestring = 10
+    max_num_segments_per_linestring = 50
+
+    points = gpd.GeoSeries(
+        [*multipoint_generator(num_pairs, max_num_points_per_multipoint)]
+    )
+    linestrings = gpd.GeoSeries(
+        [
+            *multilinestring_generator(
+                num_pairs,
+                max_num_linestring_per_multilinestring,
+                max_num_segments_per_linestring,
+            )
+        ]
+    )
+
+    d_points = cuspatial.from_geopandas(points)
+    d_linestrings = cuspatial.from_geopandas(linestrings)
+
+    result = cuspatial.pairwise_point_linestring_nearest_points(
+        d_points, d_linestrings
+    )
+    expected = compute_point_linestring_nearest_point_host(points, linestrings)
+
+    assert_geodataframe_equal(
+        result.to_pandas(),
+        expected,
+        check_dtype=False,
+        check_index_type=False,
+        check_less_precise=True,
+    )
```

Comment:
You have many failure cases in the python code that you aren't testing for here, like https://github.com/rapidsai/cuspatial/pull/681/files#diff-6e59ef32765771de56798ec48c85a59ba2bb65f4c10f6e06326a59ad82a793ffR42, also lines 56, 59, and 64.

Reply:
I've added all failure case tests except line 64 because I don't think `notimplementederror` should be tested (as it will be removed). Let me know if you insist.